tests/parallel/test_parallel.py

```diff
@@ -1,96 +1,102 @@
+        cancel_pending = kwargs.get("config", {}).get("cancel_pending", False)
+
+        with parallel(
+            thread_pool=False,
+            timeout=max_time,
+            shutdown_wait=wait_till_complete,
+            shutdown_cancel_pending=cancel_pending,
+        ) as p:
+            for test in parallel_tests:
+                p.spawn(execute, test, kwargs, results)
+                sleep(1)
+
+        # Convert results to a regular dictionary to inspect
+        results = dict(results)
+        log.info(f"Final test results: {results}")
+
+        test_rc = 0
+        for key, value in results.items():
+            log.info(f"{key} test result is {'PASS' if value == 0 else 'FAILED'}")
+            if value != 0:
+                test_rc = value
+
+        return test_rc
+
+
+def execute(test, args, results):
     """
-    Executes the test under parallel in module named 'Parallel run'  parallely.
-
-    It involves the following steps
-        - Importing of test module based on test
-        - Running the test module
+    Executes the test under parallel in module named 'Parallel run' parallely.
 
     Args:
         test: The test module which needs to be executed
-        cluster: Ceph cluster participating in the test.
-        config:  The key/value pairs passed by the tester.
-        results: results in dictionary
-
-    Returns:
-        int: non-zero on failure, zero on pass
+        args: Arguments passed to the test
+        results: Shared dictionary to store test results
     """
-
     test = test.get("test")
-    if "clusters" in test:
-        log.info("Its a multisite setup")
-        cluster = test["clusters"].get(
-            "ceph-rgw1", test["clusters"][list(test["clusters"].keys())[0]]
-        )
-        config = cluster["config"]
-    else:
-        config = test.get("config", dict())
-    config.update(args["config"])
-    file_name = test.get("module")
-    mod_file_name = os.path.splitext(file_name)[0]
-    test_mod = importlib.import_module(mod_file_name)
-    testcase_name = test.get("name", "Test Case Unknown")
-
-    rc = test_mod.run(
-        ceph_cluster=args["ceph_cluster"],
-        ceph_nodes=args["ceph_nodes"],
-        config=config,
-        test_data=args["test_data"],
-        ceph_cluster_dict=args["ceph_cluster_dict"],
-        clients=args["clients"],
+    test_name = test.get("name", "unknown_test")
+    module_name = os.path.splitext(test.get("module"))[0]
+    run_dir = args["run_config"]["log_dir"]
+    log_file = os.path.join(run_dir, f"{test_name}.log")
+
+    # Configure logger for this test
+    test_logger = Log(module_name)
+    for handler in logging.getLogger().handlers[:]:
+        logging.getLogger().removeHandler(handler)
+    logging.basicConfig(
+        handlers=[logging.FileHandler(log_file)], level=logging.INFO, format=LOG_FORMAT
     )
+    console_handler = logging.StreamHandler()
```

Comment:
By default logging to console is enabled. Do we need this?

Reply:
HI Satya
With out this parallel logs of the suites were not logging to console